sys/include/string_utils.h

```diff
@@ -38,6 +38,64 @@
 extern "C" {
 #endif
 
+/**
+ * @brief   String Writer structure.
+ *          Helper for writing multiple formatted strings to a buffer
+ */
+typedef struct {
+    const char *start;  /**< start of the target buffer */
+    char *ptr;          /**< current write pointer      */
+    size_t len;         /**< remaining length in buffer */
+} string_writer_t;
+
+/**
+ * @brief   Initialize a string writer structure
+ *
+ * @param[out]  sw      String Writer object to initialize
+ * @param[in]   data    destination buffer
+ * @param[in]   len     size of the destination buffer
+ */
+static inline void string_writer_init(string_writer_t *sw, void *data, size_t len)
+{
+    sw->start = data;
+    sw->ptr = data;
+    sw->len = len;
```

Comment:
```suggestion
    sw->len = len;
    sw->start[0] = '\0';
```

Idea: What if I have a used buffer and the user does:
```c

string_writer_init(..);
puts(string_writer_str(..));
```